src/handlers/motions/motionFinalized/helpers.ts

```diff
@@ -375,3 +390,111 @@ export const updateColonyUnclaimedStakes = async (
     });
   }
 };
+
+export const claimExpenditurePayouts = async (
+  action: string,
+  colonyAddress: string,
+): Promise<void> => {
+  const colonyClient = await getCachedColonyClient(colonyAddress);
+
+  const parsedAction = parseAction(action, [colonyClient, null, null, null]) as
```

Comment:
do we need the 3 nulls here? since the function will break if client doesn't exist
![image](https://github.com/JoinColony/block-ingestor/assets/23449297/1b124804-9faf-4a09-b075-d870ed2fd557)

Reply:
I’ve changed the type to this:

```
interface MotionActionClients {
  colonyClient?: AnyColonyClient | null;
  oneTxPaymentClient?: AnyOneTxPaymentClient | null;
  stakedExpenditureClient?: AnyStakedExpenditureClient | null;
  stagedExpenditureClient?: AnyStagedExpenditureClient | null;
}
```

Which I prefer as when calling the function it is more explicit which clients are being passed and can choose not to pass particular clients.

I just wanted to confirm that the order in which we check the clients doesn’t matter? The previous type enforced the order so that ColonyClient is checked first before the extensions. I don't think there is, but I’m not 100% sure if there could be a situation where the matching function fragment could exist on multiple clients and whether we could want to prioritise the colony client.

Reply:
i like this new type, so all good from my side :+1:

Reply:
@bassgeta what do you mean by "break"? 

@iamsamgibbs that looks like a good idea, it's now down to the developer to make sure they're passing colony client as the first property of that clients object.

Reply:
well if client was null it just didn't go through, I didn't double check the type and only later found that we need to always pass an array of length 4 in. But this new type seems clearer